lib/dl_file_uploader_worker_lib/dl_file_uploader_worker_lib/tasks/excel.py

```diff
@@ -108,9 +108,14 @@ async def run(self) -> TaskResult:
                     sheet_data_source.error = FileProcessingError.from_exception(exc.EmptyDocument())
                     sheet_data_source.status = FileProcessingStatus.failed
                 else:
-                    has_header, raw_schema, raw_schema_header, raw_schema_body = guess_header_and_schema_excel(
-                        sheetdata
-                    )
+                    try:
+                        has_header, raw_schema, raw_schema_header, raw_schema_body = guess_header_and_schema_excel(
+                            sheetdata
+                        )
+                    except Exception as ex:
+                        sheet_data_source.status = FileProcessingStatus.failed
+                        exc_to_save = ex if isinstance(ex, exc.DLFileUploaderBaseError) else exc.ParseFailed()
+                        sheet_data_source.error = FileProcessingError.from_exception(exc_to_save)
```

Comment:
What is this about? Is this the part you had to revert previously

Reply:
> What is this about? Is this the part you had to revert previously?

yes it is. when doing tests like google sheets, I need to have the same logic for sheet's processing